.github/workflows/codeql-analysis.yml

```diff
@@ -68,74 +68,89 @@ jobs:
             autoconf-archive \
             uuid-dev \
             libjansson-dev \
-            python
+            python \
+            stgit
 
     - if: matrix.language == 'cpp'
-      name: build-libnl
-      run: |
-        cd ..
-        git clone https://github.com/sonic-net/sonic-buildimage
-        pushd sonic-buildimage/src/libnl3
-        git clone https://github.com/thom311/libnl libnl3-3.5.0
-        pushd libnl3-3.5.0
-        git checkout tags/libnl3_5_0
-        git apply ../patch/0001-mpls-encap-accessors.patch
-        git apply ../patch/0002-mpls-remove-nl_addr_valid.patch
-        ln -s ../debian debian
-        fakeroot dpkg-buildpackage -us -uc -b
-        popd
-        popd
-
-    - if: matrix.language == 'cpp'
-      name: build-swss-common
+      name: Build sonic-swss-common
       run: |
         cd ..
         git clone https://github.com/sonic-net/sonic-swss-common
         pushd sonic-swss-common
         ./autogen.sh
-        fakeroot dpkg-buildpackage -us -uc -b
+        dpkg-buildpackage -rfakeroot -us -uc -b -j$(nproc)
         popd
-        dpkg-deb -x libswsscommon_1.0.0_amd64.deb $(dirname $GITHUB_WORKSPACE)
-        dpkg-deb -x libswsscommon-dev_1.0.0_amd64.deb $(dirname $GITHUB_WORKSPACE)
+        dpkg-deb -x libswsscommon_${SWSSCOMMON_VER}_amd64.deb $(dirname $GITHUB_WORKSPACE)
+        dpkg-deb -x libswsscommon-dev_${SWSSCOMMON_VER}_amd64.deb $(dirname $GITHUB_WORKSPACE)
+      env:
+        SWSSCOMMON_VER: "1.0.0"
 
     - if: matrix.language == 'cpp'
-      name: build-sairedis
+      name: Build sonic-sairedis
       run: |
         cd ..
         git clone --recursive https://github.com/sonic-net/sonic-sairedis
         pushd sonic-sairedis
         ./autogen.sh
-        DEB_BUILD_OPTIONS=nocheck SWSS_COMMON_INC="$(dirname $GITHUB_WORKSPACE)/usr/include" SWSS_COMMON_LIB="$(dirname $GITHUB_WORKSPACE)/usr/lib/x86_64-linux-gnu" fakeroot debian/rules CFLAGS="-Wno-error" CXXFLAGS="-Wno-error" binary-syncd-vs
+        DEB_BUILD_OPTIONS=nocheck \
+        SWSS_COMMON_INC="$(dirname $GITHUB_WORKSPACE)/usr/include" \
+        SWSS_COMMON_LIB="$(dirname $GITHUB_WORKSPACE)/usr/lib/x86_64-linux-gnu" \
+        DEB_CFLAGS_SET="-Wno-error" DEB_CXXFLAGS_SET="-Wno-error" \
+        dpkg-buildpackage -rfakeroot -us -uc -b -Psyncd,vs,nopython2 -j$(nproc)
         popd
+        dpkg-deb -x libsairedis_${SAIREDIS_VER}_amd64.deb $(dirname $GITHUB_WORKSPACE)
+        dpkg-deb -x libsairedis-dev_${SAIREDIS_VER}_amd64.deb $(dirname $GITHUB_WORKSPACE)
+        dpkg-deb -x libsaimetadata_${SAIREDIS_VER}_amd64.deb $(dirname $GITHUB_WORKSPACE)
+        dpkg-deb -x libsaimetadata-dev_${SAIREDIS_VER}_amd64.deb $(dirname $GITHUB_WORKSPACE)
+        dpkg-deb -x libsaivs_${SAIREDIS_VER}_amd64.deb $(dirname $GITHUB_WORKSPACE)
+        dpkg-deb -x libsaivs-dev_${SAIREDIS_VER}_amd64.deb $(dirname $GITHUB_WORKSPACE)
+      env:
+        SAIREDIS_VER: "1.0.0"
 
+    # Inject libnl deb only after sonic-sairedis compilation is done.
```

Comment:
Just to check, why should the libnl deb package be built and used only after sonic-sairedis is compiled?

Also, it looks like sonic-swss-common links to libnl. While these artifacts won't be used outside of CodeQL analysis, it does result in sonic-swss-common linking to libnl 3.4.0 while sairedis links to libnl 3.5.0.

Reply:
@saiarcot895 is it clear for you now? Basically, i didn't change the original flow.

Reply:
Hmm, ok, I'll approve it, but ideally, the same version (at least same ABI) of `libnl` should be used for all compilations. At least for CodeQL, this multiple linkage probably doesn't matter, since, as far as I know, CodeQL doesn't run the binaries.